### DISCUSSION TITLE
Keine Geräte bei Einrichtung gefunden

### DISCUSSION THREAD

**Discussion by able90** [SukramJ/aiohomematic#679] [Q&A]
Hallo zusammen,

ich versuche seit einiger Zeit die mit der neuen Homematic Integration meine CCU3 in HA einzubinden. Ich habe nach Anleitung HACS usw. installiert und auch die HA Integration erfolgreich installiert. Leider werden dann beim Hinzufügen der neuen Integration keine der gekoppelten Geräte gefunden sondern nur die CCU3 mit 24 Entitäten (Buttons und energycounter).

Randbedingungen:
- Original CCU3 Firmware 3.65.11
- Benutzer hat Adminrechte
- Firewall ausgeschaltet (testweise)
- Zugriff auf XML-RPC API auf Vollzugriff gestellt
- XML-RPC Authentifizierung eingeschaltet
- ca. 15 Geräte (Thermostate, Wandschalter, Temperatursensoren, Fensterkontakte) in der CCU eingerichtet

- Homeassistant OS läuft in einer VM
- OS und alle Integrationen, Addons sind up to date

Was ich bisher versucht habe:
- HACS -> Integration neu herunterladen
- CCU und HA neustarts
- Diverse male die Integration gelöscht und neu erstellt

Mir ist noch aufgefallen, dass ich einen unknown error erhalte wenn ich versuche "Homematic-Wired (BidCos-Wired)" einzuschalten. Das ist erstmal nicht schlimm, da ich keine Wired Geräte habe aber vielleicht hilft es ja bei der Fehlersuche.
Ansonsten dauert das Einrichten der Integration ziemlich lange (mehrere Minuten) und die CCU ist währenddessen offenbar auch ziemlich beschäftigt, sodass die Weboberfläche sich nicht mehr bedienen lässt.
In den HA Logs (System -> Logs) habe ich nichts relevantes gefunden. Gibt es irgendwo noch ausführlichere oder integrationsspezifische Logs?

Im Log der CCU scheint mir nur folgende Meldung relevant zu sein:
2022-10-25 12:55:24,188 de.eq3.cbcs.legacy.bidcos.rpc.LegacyServiceHandler INFO  [vert.x-worker-thread-4] (un)registerCallback on LegacyServiceHandler called from url: http://192.168.10.70:32795 

Die genannte IP passt zu meiner Home Assistant VM. So richtig kann ich damit aber nichts anfangen.

Ich bin mittlerweile etwas ratlos. Vielleicht habt ihr eine Idee woran es liegen kann?

Viele Grüße

Able

**Comment by SukramJ** [comment]
Interessant wäre noch die verwendeten Versionen (HA, custom_homematic) und Diagnoseinformationen(kann in der Integration heruntergeladen werden).
Kann die CCU die HA(192.168.10.70) auf Port 32795 erreichen? 
Wie ist die IP der CCU?
Welche Virtualisierung wird verwendet?

**Reply by able90** [reply]
Danke erstmal für die Antwort:
Bei HA nutze ich folgende Version:
Home Assistant 2022.10.5 
Supervisor 2022.10.0 
Operating System 9.2 
Frontend 20221010.0 - latest 

custom_homematic: 1.20.0 (ist gerade reingekommen, ändert aber nichts an dem Problem)

IP von HA: 192.168.10.70
IP der CCU: 192.168.50.60 (wird geroutet, keine Firewall dazwischen)
Die CCU kann auf jeden Fall die IP von HA erreichen. Bei dem Port bin ich nicht ganz sicher. Laut Portscanner ist er zur Zeit geschlossen. Ich habe testweise auch mal feste Callback Ports vorgegeben. Die waren laut Portscan dann geöffnet, das hat aber auch nicht geholfen. Ich habe auch keine Möglichkeit gefunden die Firewall von HA zu konfigurieren, machen die Integrationen das automatisch?

Generell ist HA aber erreichbar, die Integrationen der Hue Bridge und eines Denon AV-Receivers funktionieren (auch über Subnetzgrenzen) tadellos.

Als VM Host nutze ich Proxmox auf einem x86 Server. Installiert wurde Home Assistant OS aus einem KVM Image entsprechend der Anleitung.

Die Diagnoseinformationen habe ich angehängt.

[config_entry-homematicip_local-bd56c3c33390ad89c6dbf0ee93c163ca.json(1).txt](https://github.com/danielperna84/hahomematic/files/9861135/config_entry-homematicip_local-bd56c3c33390ad89c6dbf0ee93c163ca.json.1.txt)

**Reply by SukramJ** [reply]
Bin mir nicht sicher aber bauen Hue Bridge und Denon AV-Receiver eine eigene Verbindung zu HA auf?
Das Problem entsteht, wenn die CCU HA nicht erreichen kann. Hast Du den Verbindungstest aus der CCU raus gemacht?

> Firewall von HA zu konfigurieren

HA hat keine Firewall. Ist eine für die VM eingestellt.

> Welche Virtualisierung wird verwendet?

**Reply by able90** [reply]
Vielen Dank, daran hat es gelegen!
Es war eine Firewall vor der VM, die eingehende Verbindungen nur zugelassen hat wenn diese vom HA initiiert wurden.
Dann scheinen die Hue Bridge und der Receiver wohl keine eigenständigen Verbindungen zum HA aufzubauen.
